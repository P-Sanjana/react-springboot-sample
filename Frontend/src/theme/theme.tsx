import { Color } from "@material-ui/core";
import { createTheme } from "@material-ui/core/styles";
import GraphikMedium from "../assets/fonts/GraphikMedium.otf";
import GraphikRegular from "../assets/fonts/GraphikRegular.otf";
import GraphikSemiBold from "../assets/fonts/GraphikSemibold.otf";

type ColorPartial = Partial<Color>;

declare module "@material-ui/core/styles/createPalette" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface PaletteColor extends ColorPartial {}
}

declare module "@material-ui/core/styles/createTypography" {
  interface Typography {
    inhherit: React.CSSProperties;
    caption2: React.CSSProperties;
  }

  interface TypographyOptions {
    inherit?: React.CSSProperties;
    caption2?: React.CSSProperties;
  }
}

declare module "@material-ui/core/Typography/Typography" {
  interface TypographyPropsVariantOverrides {
    caption2: true;
  }
}

const fontFaceGraphikRegular = {
  fontFamily: "GraphikRegular",
  src: `url(${GraphikRegular}) format("opentype")`,
};

const fontFaceGraphikSemiBold = {
  fontFamily: "GraphikSemiBold",
  src: `url(${GraphikSemiBold}) format("opentype")`,
};

const fontFaceGraphikMedium = {
  fontFamily: "GraphikMedium",
  src: `url(${GraphikMedium}) format("opentype")`,
};

const theme = createTheme({
  spacing: [0, 4, 8, 12, 16, 20, 24, 32],
  palette: {
    primary: {
      main: "#0052FF",
      "100": "#FAFCFF",
      "300": "#CCE3FF",
      "500": "#0052FF",
      "700": "#002EB7",
      "900": "#00177A",
    },
    success: {
      main: "#20B03F",
      "500": "#20B03F",
      "100": "#E9F7EC",
    },
    warning: {
      main: "#FFA74F",
      "500": "#FFA74F",
      "100": "#FFF6ED",
    },
    error: {
      main: "#B71A33",
      "500": "#B71A33",
      "100": "#F3E6EB",
    },
    grey: {
      "50": "#F2F2F7",
      "100": "#E8E8F7",
      "300": "#B4B4CF",
      "500": "#4B4B60",
      "700": "#252545",
      "900": "#0E0E2E",
    },
    text: {
      primary: "#343446",
      secondary: "#7D7D89",
      hint: "#7D7D89",
      disabled: "#B2B2B9",
    },
    secondary: {
      main: "#FFA74F",
      contrastText: "#FFFFFF",
    },
  },
  overrides: {
    MuiCssBaseline: {
      "@global": {
        "@font-face": [
          fontFaceGraphikRegular,
          fontFaceGraphikSemiBold,
          fontFaceGraphikMedium,
        ],
      },
    },
  },
  typography: {
    overline: {
      gammaOverlineFontFamily: "Comic Sans Ms ",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "12px",
      lineHeight: "14px",
      identicalToBoxHeightOr117LetterSpacing: "0.005em",
      betaTextHighEmphasisColor: "#343446",
    },
    h4: {
      fontFamily: "GraphikMedium",
      fontSize: "40px",
      fontWeight: 500,
      letterSpacing: "-1%",
      lineHeight: "54px",
    },
    h6: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "24px",
      lineHeight: "34px",
    },
    caption: {
      fontFamily: "GraphikMedium",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    inherit: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    button: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "42px",
      letterSpacing: "0.01em",
    },
    body2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    subtitle2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    subtitle1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    body1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    caption2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
  },
});

export default theme;
