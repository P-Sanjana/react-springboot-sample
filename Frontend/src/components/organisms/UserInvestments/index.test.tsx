import React from "react";
import { render } from "@testing-library/react";
import bitcoin from "../../../assets/coinImage/bitcoin.png";
import theme from "../../../theme/theme";
import MyPortfolio from "./index";

const Cards = [
  {
    icon: `${bitcoin}`,
    currencyType: "Bitcoin",
    currencyCode: "BTC",
    currencyValue: "50000.00",
    gains: "+10.5",
    color: `${theme.palette.success.main}`,
  },
];

describe("MyPortfolio Component", () => {
  it("should match snapshot", () => {
    const wrapper = render(
      <MyPortfolio portfolioCards={Cards} totalBalance="$ 14,027.2" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
