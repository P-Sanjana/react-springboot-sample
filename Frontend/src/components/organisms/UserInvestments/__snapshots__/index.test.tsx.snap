// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPortfolio Component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-root-7"
      >
        <div
          class="MuiGrid-root makeStyles-parent-8 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiBox-root MuiBox-root-12 makeStyles-dashboard-1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
                style="color: rgb(52, 52, 70); font-weight: bold;"
              >
                My portfolio
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiBox-root MuiBox-root-13 makeStyles-icons-2"
            >
              <div
                class="MuiBox-root MuiBox-root-14 makeStyles-charts-3"
              >
                <img
                  alt="charts"
                  src="donut-chart-line.svg"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-15 makeStyles-menu-5"
              >
                <img
                  alt="menu"
                  class="makeStyles-menuImage-4"
                  src="list-view-icon.svg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiBox-root MuiBox-root-16 makeStyles-render-9"
          >
            <div
              class="MuiGrid-root makeStyles-cards-10 MuiGrid-item"
            >
              <div
                class="MuiBox-root MuiBox-root-21 makeStyles-root-17"
              >
                <img
                  class="makeStyles-currencyIcon-18"
                  src="bitcoin.png"
                />
                <div
                  class="MuiBox-root MuiBox-root-22"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Bitcoin
                  </p>
                  <span
                    class="MuiTypography-root makeStyles-currencyCode-19 MuiTypography-caption"
                  >
                    BTC
                  </span>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-23"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                    data-testid="coin"
                  >
                    50000.00
                  </p>
                  <span
                    class="MuiTypography-root makeStyles-gainsText-20 MuiTypography-caption"
                    data-testid="gains"
                    style="color: rgb(32, 176, 63);"
                  >
                    +10.5
                    %
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-totalBalance-6 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
              style="color: rgb(125, 125, 137);"
            >
              Total Balance
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              $ 14,027.2
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-root-7"
    >
      <div
        class="MuiGrid-root makeStyles-parent-8 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-12 makeStyles-dashboard-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
              style="color: rgb(52, 52, 70); font-weight: bold;"
            >
              My portfolio
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-13 makeStyles-icons-2"
          >
            <div
              class="MuiBox-root MuiBox-root-14 makeStyles-charts-3"
            >
              <img
                alt="charts"
                src="donut-chart-line.svg"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-15 makeStyles-menu-5"
            >
              <img
                alt="menu"
                class="makeStyles-menuImage-4"
                src="list-view-icon.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiBox-root MuiBox-root-16 makeStyles-render-9"
        >
          <div
            class="MuiGrid-root makeStyles-cards-10 MuiGrid-item"
          >
            <div
              class="MuiBox-root MuiBox-root-21 makeStyles-root-17"
            >
              <img
                class="makeStyles-currencyIcon-18"
                src="bitcoin.png"
              />
              <div
                class="MuiBox-root MuiBox-root-22"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Bitcoin
                </p>
                <span
                  class="MuiTypography-root makeStyles-currencyCode-19 MuiTypography-caption"
                >
                  BTC
                </span>
              </div>
              <div
                class="MuiBox-root MuiBox-root-23"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                  data-testid="coin"
                >
                  50000.00
                </p>
                <span
                  class="MuiTypography-root makeStyles-gainsText-20 MuiTypography-caption"
                  data-testid="gains"
                  style="color: rgb(32, 176, 63);"
                >
                  +10.5
                  %
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-totalBalance-6 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
            style="color: rgb(125, 125, 137);"
          >
            Total Balance
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            $ 14,027.2
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
